test/integration-test-main.sh

```diff
@@ -397,6 +397,15 @@ function test_external_modification {
 function test_external_creation {
     describe "Test external creation of an object ..."
     local OBJECT_NAME; OBJECT_NAME=$(basename "${PWD}")/"${TEST_TEXT_FILE}"
+
+    # [NOTE]
+    # This test requires that "TEST_TEXT_FILE" does not exist.
+    # (If it exists, it is forgetting to remove it somewhere.)
+    # Also, if noobj_cache is enabled, do the following test first to
+    # ensure that noobj_cache caches it.
+    #
+    [ ! -e "${TEST_TEXT_FILE}" ]
+
     echo "data" | aws_cli s3 cp - "s3://${TEST_BUCKET_1}/${OBJECT_NAME}"
     # shellcheck disable=SC2009
     if ! ps u -p "${S3FS_PID}" | grep -q disable_noobj_cache; then
```

Comment:
The following line is the one that fails.  I think that this test is the same as before but since we now use a different directory for every function that it has a different behavior than before?

Reply:
Sorry -- tests use different _files_ for every run but do not `cd` into a new directory.

Reply:
I'm sorry, I'm a little unsure what's wrong.
The meaning of testing TEST_TEXT_FILE at the beginning of this test is to cache the non-existence of the file.
(If disable_noobj_cache is not specified, that cache will be needed on the next line)

Reply:
There are cases where the cache expiration date has passed, and it seems that it may fail.
I try to fix it.

Reply:
`echo "data" | aws_cli s3 cp - "s3://${TEST_BUCKET_1}/${OBJECT_NAME}"`
This process seems to take more than 700ms.
Depending on the situation, it seems to take more than 1 second.
In that case, the test here will fail.
Therefore, when testing noobj_cache, the results will be unstable.

Reply:
When noobj_cache was enabled, this test was checking that the file was cached for non-existence.
However, it depends on the processing time of aws cli as mentioned above.
The processing time may take more than 1 second depending on the situation, causing an error.

I have concluded that this cannot be avoided.
As a result, it has been fixed so that it does not check that the file is cached for non-existence.